homeassistant/components/sensor/discogs.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_TOKEN)
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:album'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+})
+
+REQUIREMENTS = ['discogs_client==2.2.1']
+
+SCAN_INTERVAL = timedelta(seconds=7200)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the discogs sensor."""
+    token = config.get(CONF_TOKEN)
+
+    async_add_devices([DiscogsCollection(token)], True)
```

Comment:
What happens if the token is not valid? I assume that there will be a traceback.

Reply:
Ok, I'm not at all sure how to do this, but I'll give it a try someday this week!

Reply:
See DiscogsCollection#__init__ which returns false if the user is not present

Reply:
I guess that with an invalid token a `discogs_client.exceptions.HTTPError` will be thrown. This will happen before init return. The whole setup of the discogs client should happen in `setup()`.

Reply:
I'll try and check again but it probably won't be anytime soon.